### ISSUE TITLE
Problem with runtime loading of another npm libs bundle

### ISSUE THREAD

**Issue by djaonourside** [elsassph/haxe-modular#107]
Currently when besides static _libs.js_, I dynamically load another _libs2.js_ with npm libs,` require('some-lib-from-libs2') `doesn't find it because `require` function from index.js returns cashed __registry __ from libs.js

**Comment by elsassph** [comment]
Is that a node app?

**Comment by djaonourside** [comment]
@elsassph No, it's a browser app. Firstly libs.js and index.js are loaded with index.html.  Then lib2.js is loaded in runtime. And finally a js file loaded with require(''some-lib-from-libs2'').

**Comment by elsassph** [comment]
Oh do you mean there are 2 Modular apps loaded in one page? This won't work right now - there is no scoping of the global brigde...

**Comment by djaonourside** [comment]
I see. Is it planned to implement such feature? One libs file is sometimes too heavy :)

**Comment by elsassph** [comment]
Hmm wait. It should work to have 2 libs, or at least you can make it work by making sure the code in the `libs2.js` adds libraries to the global `__registry__` object.

**Comment by elsassph** [comment]
The libs template here https://github.com/elsassph/haxe-modular/blob/master/doc/npm-setup.md
should allow multiple libs.js to add to the registry.

**Comment by djaonourside** [comment]
`var require = (function(r){ return function require(m) { return r[m]; } })($s.__registry__ || {});`
the variable `require` is inited when index.js runs and r variable in the function stays equals __register__ from libs.js and doesn't change when libs2.js has been loaded.

**Comment by elsassph** [comment]
Oooh. Ok my bad, so the pattern in the doc is indeed wrong. The solution would be to "add" the libraries references into the existing `__registry__` instead of replacing the global object.

**Comment by elsassph** [comment]
But it would be better to always resolve this global registry instead of keeping a reference to the first one found.

**Comment by peteshand** [comment]
@elsassph I'm running into the same issue.
How about changing the following line in split.js from:
`buffer += "var require = (function(r){ return function require(m) { return r[m]; } })($s.__registry__ || {});\n";`
to
`buffer += "var require = (function(){ return function require(m) { return $s.__registry__[m]; } })();\n";`

If it looks good let me know and I'll submit a pull request

**Comment by elsassph** [comment]
@peteshand you're right, this looks like I did a mistake here.

**Comment by peteshand** [comment]
Cool, I've created a pull request: https://github.com/peteshand/haxe-modular/pull/1
